code-gen/src/main/scala/scalapb/validate/NumericRulesPrinter.scala

```diff
@@ -0,0 +1,504 @@
+package scalapb.validate
+
+import com.google.protobuf.Descriptors.FieldDescriptor
+import io.envoyproxy.pgv.validate.Validate
+
+object NumericRulesPrinter {
+
+  // constant definition
+  private val COMPARATIVE_VALIDATION = "io.envoyproxy.pgv.ComparativeValidation"
+  private val CONSTANT_VALIDATION = "io.envoyproxy.pgv.ConstantValidation"
+
+  private val JAVA_INT_PKG = "java.lang.Integer"
+  private val JAVA_LONG_PKG = "java.lang.Long"
+  private val JAVA_FLOAT_PKG = "java.lang.Float"
+  private val JAVA_DOUBLE_PKG = "java.lang.Double"
+
+  private def gt[T](
```

Comment:
Read the second comment first - but I believe this set of functions can also be deduped since all the difference is just the method name they call on COMPARATIVE_VALIDATION.